src/coreclr/jit/emitxarch.cpp

```diff
@@ -1034,25 +1030,26 @@ bool emitter::TakesEvexPrefix(const instrDesc* id) const
 {
     instruction ins = id->idIns();
 
-    if (!IsEvexEncodedInstruction(ins))
+    if (!IsEvexEncodableInstruction(ins))
     {
+        // Can't be EVEX encoded ever
         return false;
     }
 
-    if (!emitComp->DoJitStressEvexEncoding())
+    if (HasHighSIMDReg(id) || (id->idOpSize() == OPSZ64) || HasKMaskRegisterDest(ins))
```

Comment:
The `IsEvexEncodableInstruction` excludes the kmask instructions, since they are VEX only. This simplifies the checks we do here since it means we can just check for kmask usage in general.